wordpress-mobile/WordPress-Android
# Fix a potential issue related to lifecycle

A follow-up PR for #20937

As @thomashorta mentioned in #20937, the previous implementation causes some potential issues. This PR also rename the helper class to emphasize the usage is related to I/O.

-----

## To Test:

1. Sign in JP app.
2. Go to the Reader tab.
3. Click on a post.
4. Click on a tag in that post.
5. Click Subscribe/unsubscribe button on the tag header.
6. It should work as usual.
7. Done, thank you!

-----

## Regression Notes

1. Potential unintended areas of impact

    - reader

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - manual

3. What automated tests I added (or what prevented me from doing so)

    - n/a

-----

## PR Submission Checklist:

Skipped

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones):

Skipped
